apps/chat/src/store/publication/publication.reducers.ts

```diff
@@ -0,0 +1,162 @@
+import { PayloadAction, createSlice } from '@reduxjs/toolkit';
+
+import { ApiKeys, UploadStatus } from '@/src/types/common';
+import {
+  Publication,
+  PublicationInfo,
+  PublicationRule,
+} from '@/src/types/publication';
+
+import * as PublicationSelectors from './publication.selectors';
+
+export { PublicationSelectors };
+
+export interface PublicationState {
+  publications: (PublicationInfo & Partial<Publication>)[];
+  selectedPublication: Publication | null;
+  resourcesToReview: {
+    publicationUrl: string;
+    reviewed: boolean;
+    reviewUrl: string;
+  }[];
+  rules: Record<string, PublicationRule[]>;
+  isRulesLoading: boolean;
+}
+
+const initialState: PublicationState = {
+  publications: [],
+  selectedPublication: null,
+  resourcesToReview: [],
+  rules: {},
+  isRulesLoading: false,
+};
+
+export const publicationSlice = createSlice({
+  name: 'publication',
+  initialState,
+  reducers: {
+    init: (state) => state,
+    publish: (
+      state,
+      _action: PayloadAction<{
+        resources: { sourceUrl: string; targetUrl: string }[];
+        targetFolder: string;
+        rules: PublicationRule[];
+      }>,
+    ) => state,
+    publishFail: (state) => state,
+    uploadPublications: (state) => state,
+    uploadPublicationsSuccess: (
+      state,
+      { payload }: PayloadAction<{ publications: PublicationInfo[] }>,
+    ) => {
+      state.publications = payload.publications;
+    },
+    uploadPublicationsFail: (state) => state,
+    uploadPublication: (state, _action: PayloadAction<{ url: string }>) =>
+      state,
+    uploadPublicationSuccess: (
+      state,
+      { payload }: PayloadAction<{ publication: Publication }>,
+    ) => {
+      state.publications = state.publications.map((p) =>
+        p.url === payload.publication.url
+          ? { ...payload.publication, ...p, uploadStatus: UploadStatus.LOADED }
+          : p,
+      );
+    },
+    uploadPublicationFail: (state) => state,
+    deletePublication: (
+      state,
+      _action: PayloadAction<{
+        targetFolder: string;
+        resources: { targetUrl: string }[];
+      }>,
+    ) => state,
+    deletePublicationFail: (state) => state,
+    uploadPublishedWithMeItems: (
+      state,
+      _action: PayloadAction<{ featureType: ApiKeys }>,
```

Comment:
Use `FeatureType` instead
```suggestion
      _action: PayloadAction<{ featureType: FeatureType }>,
```
and `getApiKeyByFeatureType` in `getPublishedWithMeItems`

Reply:
This is where I was rather misleading. I changed the name of the variable. Why pass featureType to get apikey if we can just pass apiKey?